### PR TITLE
ML-164 normalised reused code

### PR DESCRIPTION
**Pull request by MarkusVK**

### CODE REVIEW

File: core/models/model_training.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
+import torch
+import copy
+import sys
+from tqdm.notebook import tqdm
+from torch.utils.data import DataLoader
+import torch.nn as nn
+from core.util.early_stop import EarlyStop
+
+learning_rate = 0.001
```
Comment (Tinggaard) [inline core/models/model_training.py:9]: Why is this parameter defined in this scope, if it is only one function that uses it?

File: core/models/model_training.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    """Train a model."""
+    optimizer = torch.optim.Adam(model.parameters(), lr=learning_rate)
+
+    best_v_loss = sys.maxsize
+    best_model = None
+
+    early_stopper = EarlyStop(5, 0.05)
+
+    plot_train_loss = []
+    plot_val_loss = []
```
Comment (Tinggaard) [inline core/models/model_training.py:57]: Omit the `plot_` prefix, as they are not used for plotting in this context.

File: core/models/GRU.py
Comment (Tinggaard) [inline core/models/GRU.py:1]: Maybe we should introduce a `__init__` file for the `models` directory, such that we can import all the models directly from this.

File: core/notebooks/01_trefor-park-gru.ipynb
Comment (Tinggaard) [inline core/notebooks/01_trefor-park-gru.ipynb:1]: You would be a skattebasse if you changed the names of all the notebooks to a single naming schema that makes sense.

Could be something like:
`<experiment-number>-<model>-<dataset>.ipynb`
Ie
`0-GRU-trefor-park`
`0-LSTM-trefor-park`
And then later we would have
`1-GRU-trefor-park`
`1-LSTM-trefor-park`

Maybe this should be another PR, if we do gridsearch on all of the experiments and that is not implemented in these notebooks - just a thought though.

**Review by Tinggaard — CHANGES_REQUESTED** [review]
image

**Comment by Tinggaard** [comment]
@MarkusVK if you create the file `core/models/__init__.py` and put the following in it:
```python
"""All the models for ZAP."""

from .GRU import GRU
from .LSTM import LSTM

__all__ = ["GRU", "LSTM"]
```

Then, we can import as follows:
```python
from core.models import GRU, LSTM
```

**Comment by Tinggaard** [comment]
nevermind lol

**Review by Tinggaard — APPROVED** [review]
